module_utils/ansible_nailgun_cement.py

```diff
@@ -138,3 +162,29 @@ def parse_template_from_file(file_name, module):
     except Exception as e:
         module.fail_json(msg='Error while reading template file: ' + str(e))
     return template_dict
+
+
+def find_organization(module, name, failsafe=False):
+    org = Organization(name=name)
+    response = org.search(set(), {'search': 'name="{}"'.format(name)})
+    return handle_find_response(module, response, message="No organization found for %s" % name, failsafe=failsafe)
+
+
+def find_lifecycle_environment(module, name, organization, failsafe=False):
+    response = entities.LifecycleEnvironment(name=name, organization=organization).search()
+    return handle_find_response(module, response, message="No lifecycle environment found for %s" % name, failsafe=failsafe)
+
+
+def handle_find_response(module, response, message="", failsafe=False):
+    if len(response) == 1:
+        return response[0]
+    elif failsafe:
+        return None
+    else:
+        module.fail_json(msg=message)
+
```

Comment:
Fair enough, I understand your point. Writing a unity-version is definitely challenging.
However changing all single functions in case of an API change could be a lot of work as well ;)